plugins/inputs/mongodb/mongodb.go

```diff
@@ -110,7 +116,11 @@ func (m *MongoDB) Init() error {
 
 		err = client.Ping(ctx, opts.ReadPreference)
 		if err != nil {
-			return fmt.Errorf("unable to connect to MongoDB: %s", err)
+			if !m.IgnoreUnreachableHosts {
+				return fmt.Errorf("unable to connect to MongoDB: %s", err)
```

Comment:
Errors should be printed with `%w`
```suggestion
				return fmt.Errorf("unable to connect to MongoDB: %w", err)
```